[Flaky] failed to delete cluster "kind"

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->

/kind flake


**What happened**:

```
Switched to context "kind-kind".
Deleting cluster "kind" ...

ERROR: failed to delete cluster "kind": failed to delete nodes: command "docker rm -f -v kind-worker kind-control-plane kind-worker2" failed with error: exit status 1

Command Output: kind-control-plane
Error response from daemon: cannot remove container "kind-worker": could not kill container: tried to kill container, but did not receive an exit event
Error response from daemon: cannot remove container "kind-worker2": could not kill container: tried to kill container, but did not receive an exit event
make: *** [Makefile-test.mk:176: run-test-e2e-singlecluster-1.32.11] Error 1
```

**What you expected to happen**:

No errors

**How to reproduce it (as minimally and precisely as possible)**:

https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_kueue/8893/pull-kueue-test-e2e-main-1-32/2017169874874273792

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/hack/testing/e2e-common.sh
+++ b/hack/testing/e2e-common.sh
@@ -160,8 +160,8 @@ export E2E_TEST_AGNHOST_IMAGE=${E2E_TEST_AGNHOST_IMAGE_WITH_SHA%%@*}
 # $1 cluster name
 function cluster_cleanup {
     local cluster_name="$1"
-    local max_retries=3
-    local retry_delay=10
+    local max_retries=5
+    local retry_delay=1
 
     for attempt in $(seq 1 "$max_retries"); do
         if $KIND delete cluster --name "$cluster_name"; then
@@ -174,17 +174,12 @@ function cluster_cleanup {
 
         echo "WARNING: kind delete cluster '$cluster_name' failed (attempt $attempt/$max_retries)."
 
-        # Force-stop Docker containers before retrying. Works around a
-        # Docker/containerd race condition where 'docker rm -f' fails with
-        # "could not kill container ... did not receive an exit event".
-        # See https://github.com/kubernetes-sigs/kueue/issues/8908
-        echo "Forcing Docker container cleanup for cluster '$cluster_name'..."
-        local containers
-        containers=$(docker ps -aq --filter "label=io.x-k8s.kind.cluster=${cluster_name}" 2>/dev/null || true)
-        for cid in $containers; do
-            docker kill "$cid" 2>/dev/null || true
-            docker rm -f -v "$cid" 2>/dev/null || true
-        done
+        # Retry kind deletion to work around Docker race conditions where
+        # delete can fail before receiving the container exit event.
+        # Upstream bugs:
+        # - https://github.com/moby/moby/issues/51028
+        # - https://github.com/moby/moby/issues/51845
+        # This workaround can be revisited after upstream fixes are available.
 
         echo "Retrying in ${retry_delay}s..."
         sleep "$retry_delay"

